app/models/comment.php

```diff
@@ -50,4 +60,73 @@ public function write($thread_id)
 
         $db->insert('comment', $params);
     }
+
+    public static function sortMostComments()
+    {
+        $db = DB::conn();
+
+        $thread_id = $db->rows('SELECT COUNT(*), thread_id FROM comment GROUP BY thread_id ORDER BY COUNT(*) DESC');
+
+        return $thread_id;
+    }
+
+    public static function isOwner($user_id, $id)
+    {
+        $db = DB::conn();
+
+        $row = $db->row('SELECT * FROM comment WHERE user_id = ? AND id = ?',
+        array($user_id, $id));
+
+        return (bool) $row;
+    }
+
+    public function get($comment_id)
+    {
+        if (!$this->validate()) {
+            throw new ValidationException('Invalid comment');
+        }
+
+        $db = DB::conn();
+
+        $row = $db->row('SELECT * FROM comment WHERE id = ?', array($comment_id));
+
+        if (!$row) {
+            throw new RecordNotFoundException('No record found!');
+        }
+
+        return new self($row);
+    }
+
+    public function edit()
+    {
+        if (!$this->validate()) {
+            throw new ValidationException('Invalid Comment');
+        }
+
+        $db = DB::conn();
+
+        $params = array(
+            'body' => $this->body,
+            'date_modified' => date('Y-m-d H:i:s')
+        );
+
+        $db->update('comment', $params, array('id' => $this->id));
+    }
+
+    public static function deleteAllCommentsInThread($thread_id)
+    {
+        $db = DB::conn();
+        $db->query('DELETE FROM comment WHERE thread_id = ?', array($thread_id));
+    }
+
+    public static function delete($id)
```

Comment:
delete() will only delete the comment based on its id. deleteAllCommentsInThread() will delete all comments in a thread